core/services/flux_monitor.go

```diff
@@ -73,12 +74,20 @@ func (fm *concreteFluxMonitor) Start() error {
 
 	go fm.actionConsumer(fm.ctx)
 
-	var merr error
+	count := 0
+	errChan := make(chan error)
 	err := fm.store.Jobs(func(j *models.JobSpec) bool {
-		err := fm.AddJob(*j)
-		merr = multierr.Combine(merr, err)
+		go func(j *models.JobSpec) {
+			errChan <- fm.AddJob(*j)
+		}(j)
+		count++
 		return true
 	}, models.InitiatorFluxMonitor)
+
+	var merr error
+	for i := 0; i < count; i++ {
```

Comment:
This is just making a list of the error returned by `Jobs` (which is just a single error) with the size of the number of jobs - is that really what you want? e.g: If you can't connect to the DB, you'll get:

    ["error failed to connect", "error failed to connect", "error failed to connect", ...]

Reply:
The error here should have been taken from the error channel mentioned above